aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -103,6 +102,14 @@
         Objects.requireNonNull(onSubscriptionError);
         Objects.requireNonNull(onSubscriptionComplete);
 
+        if (subscriptionUrl == null) {
+            try {
+                subscriptionUrl = buildConnectionRequestUrl();
+            } catch (ApiException apiException) {
+                onSubscriptionError.accept(apiException);
```

Comment:
Should you add `return;` after this line, to cede the flow of control from `requestSubscription(...)`?

Reply:
Yikes...good eye on that one. I debated just making the function throw, but it started betting messy upstream.